src/keystore.js

```diff
@@ -191,13 +146,16 @@ class Keystore {
 
   getPublic(keys, options = {}) {
     const formats = ['hex', 'buffer']
-    const decompress = options.decompress || true
-    const format = formats[options.format || 'hex']
+    const decompress = typeof options.decompress === "undefined" ? true : options.decompress
+    const format = options.format || 'hex';
+    if(formats.indexOf(format) === -1) {
+      throw new Error('Supported formats are `hex` and `buffer`')
+    }
     let pubKey = keys.public.marshal()
     if (decompress) {
       pubKey = secp256k1.publicKeyConvert(pubKey, false)
     }
-    return format === 'buffer' ? pubKey : pubKey.toString('hex')
+    return options.format === 'buffer' ? pubKey : pubKey.toString('hex')
```

Comment:
Why change to `options.format`?

Reply:
I edited this function a but simply because the format option wasn't actually being applied! I was able to just pas options.format downward instead of assigning it to format. You can see the passing tests now.

Reply:
Tests passing for me without the`options.` Either or is fine imo but not both

Reply:
Np! Restored `format`